src/devices/src/virtio/vsock/device.rs

```diff
@@ -217,6 +219,25 @@ where
 
         have_used
     }
+
+    pub fn write_element_in_queue(&self, idx: usize, val: u64) -> result::Result<(), DeviceError> {
+        if idx > self.queue_events.len() {
+            return Err(DeviceError::QueueError(QueueError::DescIndexOutOfBounds(
+                idx as u16,
+            )));
+        }
+        self.queue_events[idx].write(val).unwrap();
+        Ok(())
+    }
+
+    pub fn get_element_from_queue(&self, idx: usize) -> result::Result<u64, DeviceError> {
+        if idx > self.queue_events.len() {
+            return Err(DeviceError::QueueError(QueueError::DescIndexOutOfBounds(
+                idx as u16,
+            )));
+        }
+        Ok(self.queue_events[idx].as_raw_fd() as u64)
+    }
```

Comment:
Would be nice if we could have this in `vsock::test_utils` since they're not used in release code.

Reply:
These functions were added in this PR?